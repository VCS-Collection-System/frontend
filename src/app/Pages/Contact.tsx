import React from 'react';
import { Button, Card, CardBody, CardFooter } from '@patternfly/react-core';
import { ExternalLinkAltIcon } from '@patternfly/react-icons';
import CardWithTitle from '@app/Components/CardWithTitle';
import { useHistory } from 'react-router-dom';

const Contact: React.FunctionComponent = () => {
  const history = useHistory();
  return (
    <Card isRounded>
      <CardWithTitle
        title="Help"
        info="If you need more information, use the links below."
      />
      <CardBody>
        <Button
          icon={<ExternalLinkAltIcon />}
          variant="link"
          component="a"
          href="https://redhat.service-now.com/help?id=kb_article_view&sysparm_article=KB0015661"
          target="_blank"
        >
          VCS User Guide
        </Button>
        <br />
        <Button
          icon={<ExternalLinkAltIcon />}
          variant="link"
          component="a"
          href="https://redhat.service-now.com/help?id=kb_search&spa=1&workflow_state=published&kb_knowledge_base=9f106876db5b8890ab8fe8ca4896199c"
          target="_blank"
        >
          Frequently asked questions (FAQs) and related knowledge base articles
        </Button>
        <br />
        <Button
          icon={<ExternalLinkAltIcon />}
          variant="link"
          component="a"
          href="https://redhat.service-now.com/help?id=sc_cat_item&sys_id=89e35ece1b4074102d12c880604bcb6a"
          target="_blank"
        >
          Contact People Helpdesk for process questions and systems issues
        </Button>
        <br />
        <Button
          icon={<ExternalLinkAltIcon />}
          variant="link"
          component="a"
          href="https://www.redhat.com/en/services/consulting"
          target="_blank"
        >
          Developed by the Red Hat NAPS Services Team
        </Button>

      </CardBody>
      <CardFooter>
        <Button variant="plain" onClick={() => history.push("/")}>
          Back
        </Button>
      </CardFooter>
    </Card>
  );
};

export { Contact };
