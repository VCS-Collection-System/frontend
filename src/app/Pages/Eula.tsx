import React from 'react';
import {
  Button, 
  Card,
  CardFooter,
  CardBody,
  Stack,
  Text,
} from '@patternfly/react-core';
import { CheckCircleIcon } from '@patternfly/react-icons'
import CardWithTitle from '@app/Components/CardWithTitle';

const Eula = ({login}) => {

  return (     
    <Card isRounded>
      <CardWithTitle title="Welcome" info="VCS Privacy Notice" />
      <CardBody>
        <Stack hasGutter>
          <Text>
           Please review the <a href="https://source.redhat.com/departments/legal/globallegalcompliance/compliance_folder/vcs_privacy_notice_for_red_hat_globalpdf" target="_blank"  rel="noreferrer">Covid-19 and VCS Privacy Notice available on The Source</a> for information on how Red Hat collects, uses, stores, transfers, and otherwise processes personal information about you in connection with COVID-19 and the validation check-in system (VCS) for Red Hat.
          </Text>
        </Stack>
      </CardBody>
      <CardFooter>
        <Button
          icon={<CheckCircleIcon />}
          variant="primary"
          onClick={login}>
          Accept
        </Button>
      </CardFooter>
    </Card>  
  );
};

export { Eula };
