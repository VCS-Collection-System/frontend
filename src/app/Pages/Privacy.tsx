import React from 'react';
import { Button, Card, CardBody, CardFooter } from '@patternfly/react-core';
import { ExternalLinkAltIcon } from '@patternfly/react-icons';
import CardWithTitle from '@app/Components/CardWithTitle';
import { useHistory } from 'react-router-dom';

const Privacy: React.FunctionComponent = () => {
  const history = useHistory();
  return (
    <Card isRounded>
      <CardWithTitle
        title="Privacy"
        info="Covid-19 and VCS Privacy Notice is available on The Source. Please review the link below for more information on how Red Hat collects, uses, stores, transfers, and otherwise processes personal information about you in connection with COVID-19 and the validation check-in system (VCS) for Red Hat."
      />
      <CardBody>
        <Button
          icon={<ExternalLinkAltIcon />}
          variant="link"
          component="a"
          href="https://source.redhat.com/departments/legal/globallegalcompliance/compliance_folder/vcs_privacy_notice_for_red_hat_globalpdf"
          target="_blank"
        >
          Covid-19 and VCS Privacy Notice for Red Hat
        </Button>
      </CardBody>
      <CardFooter>
        <Button variant="plain" onClick={() => history.push("/")}>
          Back
        </Button>
      </CardFooter>
    </Card>
  );
};

export { Privacy };
