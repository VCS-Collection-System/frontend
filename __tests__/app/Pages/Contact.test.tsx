import React from 'react';
import { shallow, mount } from 'enzyme';
import { Contact } from '../../../src/app/Pages/Contact';

// mock history.push
import reactRouterDom from 'react-router-dom'
jest.mock('react-router-dom')
const mockPush = jest.fn()
reactRouterDom.useHistory = jest.fn().mockReturnValue({push: mockPush})

describe('Contact Page', ()=> {
    it('should be defined', () => {
        expect(Contact).toBeDefined()
    })
    it('should render without crashing', () => {
        const wrapper = shallow(<Contact />)
        expect(wrapper)
    })
    it('should match existing snapshot', () => {
        const wrapper = shallow(<Contact />)
        expect(wrapper).toMatchSnapshot()
    })

    it('Back button pushes to home page', async () => {
        const contact = mount( <Contact />)
        contact.find('button').simulate('click')
        await Promise.resolve()
        expect(mockPush).toHaveBeenCalledTimes(1)
        expect(mockPush).toHaveBeenCalledWith('/')
      })
})
