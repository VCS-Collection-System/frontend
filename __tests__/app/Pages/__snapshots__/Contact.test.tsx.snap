// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Page should match existing snapshot 1`] = `
<Card
  isRounded={true}
>
  <CardWithTitle
    info="If you need more information, use the links below."
    title="Help"
  />
  <CardBody>
    <Button
      component="a"
      href="https://redhat.service-now.com/help?id=kb_article_view&sysparm_article=KB0015661"
      icon={
        <ExternalLinkAltIcon
          color="currentColor"
          noVerticalAlign={false}
          size="sm"
        />
      }
      target="_blank"
      variant="link"
    >
      VCS User Guide
    </Button>
    <br />
    <Button
      component="a"
      href="https://redhat.service-now.com/help?id=kb_search&spa=1&workflow_state=published&kb_knowledge_base=9f106876db5b8890ab8fe8ca4896199c"
      icon={
        <ExternalLinkAltIcon
          color="currentColor"
          noVerticalAlign={false}
          size="sm"
        />
      }
      target="_blank"
      variant="link"
    >
      Frequently asked questions (FAQs) and related knowledge base articles
    </Button>
    <br />
    <Button
      component="a"
      href="https://redhat.service-now.com/help?id=sc_cat_item&sys_id=89e35ece1b4074102d12c880604bcb6a"
      icon={
        <ExternalLinkAltIcon
          color="currentColor"
          noVerticalAlign={false}
          size="sm"
        />
      }
      target="_blank"
      variant="link"
    >
      Contact People Helpdesk for process questions and systems issues
    </Button>
    <br />
    <Button
      component="a"
      href="https://www.redhat.com/en/services/consulting"
      icon={
        <ExternalLinkAltIcon
          color="currentColor"
          noVerticalAlign={false}
          size="sm"
        />
      }
      target="_blank"
      variant="link"
    >
      Developed by the Red Hat NAPS Services Team
    </Button>
  </CardBody>
  <CardFooter>
    <Button
      onClick={[Function]}
      variant="plain"
    >
      Back
    </Button>
  </CardFooter>
</Card>
`;
